resources/js/features/forums/components/ShortcodeRenderer/ShortcodeRenderer.test.tsx

```diff
@@ -0,0 +1,168 @@
+import {
+  persistedAchievementsAtom,
+  persistedGamesAtom,
+  persistedTicketsAtom,
+  persistedUsersAtom,
+} from '@/features/forums/state/forum.atoms';
+import { render, screen } from '@/test';
+import {
+  createAchievement,
+  createGame,
+  createSystem,
+  createTicket,
+  createUser,
+} from '@/test/factories';
+
+import { ShortcodeRenderer } from './ShortcodeRenderer';
+
+describe('Component: ShortcodeRenderer', () => {
+  it('renders without crashing', () => {
+    // ARRANGE
+    const { container } = render(<ShortcodeRenderer body="" />);
+
+    // ASSERT
+    expect(container).toBeTruthy();
+  });
+
+  it('given a body with a URL tag, renders the URL shortcode component', () => {
+    // ARRANGE
+    const body = '[url]https://example.com[/url]';
+
+    render(<ShortcodeRenderer body={body} />);
+
+    // ASSERT
+    expect(screen.getByTestId('url-embed-https://example.com/')).toBeVisible();
+  });
+
+  it('given a body with a self-closing URL tag, renders the URL shortcode component', () => {
+    // ARRANGE
+    const body = '[url=https://example.com]';
+
+    render(<ShortcodeRenderer body={body} />);
+
+    // ASSERT
+    expect(screen.getByTestId('url-embed-https://example.com/')).toBeVisible();
+  });
+
+  it('given a body with a code tag, renders the code shortcode component', () => {
+    // ARRANGE
+    const body = '[code]this is some stuff inside code tags[/code]';
+
+    render(<ShortcodeRenderer body={body} />);
+
+    // ASSERT
+    const textEl = screen.getByText(/this is some stuff/i);
+
+    expect(textEl).toBeVisible();
+    expect(textEl.nodeName).toEqual('PRE');
+    expect(textEl).toHaveClass('codetags');
+  });
+
+  it('given a body with a spoiler tag, renders the spoiler shortcode component', () => {
+    // ARRANGE
+    const body = '[spoiler]this is a spoiler![/spoiler]';
+
+    render(<ShortcodeRenderer body={body} />);
+
+    // ASSERT
+    expect(screen.getByRole('button', { name: /spoiler/i })).toBeVisible();
+  });
+
+  it('given a body with an img tag, renders the img shortcode component', () => {
+    // ARRANGE
+    const body = '[img]https://i.imgur.com/ov30jeD.jpg[/img]';
```

Comment:
Not sure if it matters, but using an `[img=` with a body outputs the body now.
```
[img=http://url/image.png]Image[/img]
```
Old:
![image](https://github.com/user-attachments/assets/00dc560d-0fd6-4ab3-8478-3042669b4eb1)

New:
![image](https://github.com/user-attachments/assets/b7f57ce8-7b07-4167-9ba5-6f9c0af0cd07)

Reply:
Definitely bothers me - good catch. Fixed in latest.